src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.cs

```diff
@@ -32,7 +32,7 @@ internal abstract partial class CSharpSelectionResult(
     {
         public static async Task<CSharpSelectionResult> CreateAsync(
             SemanticDocument document,
-            SelectionInfo selectionInfo,
+            FinalSelectionInfo selectionInfo,
```

Comment:
broke this into two type InitialSelectionInfo and FinalSelectionInfo so that we can make the variables clearer and so that later parts of EM don't use data they shouldn't.